app/controllers/user_counters_controller.rb

```diff
@@ -21,12 +23,23 @@ def index
   protected
 
   def permitted_params
-    params.require(:data).require(:attributes).permit(:increment)
+    params.require(:data).require(:attributes).permit(:increment, values: [])
   end
 
   def counter_name
     counter_name = params[:id]
     counter_name += ".#{params[:format]}" if params[:format].present?
     counter_name
   end
+
+  # todo: move to model
+  def apply_values(counter, values)
+    values_before = counter.values
+    new_values = values - values_before
```

Comment:
Array subtraction ensures that all `values` will be unique, which you want. However, when I read GT-1314 it also mentions

> We discard any new values that would come before the first current value when alpha sorting the list

and

> We then delete any values that come before a 'deleteBefore' value passed to the API, keeping at least 1 value

I don't see those things done here.

Reply:
@frett I believe we decided that those requirements weren't actually required for this? Should the JIRA task be updated?

Reply:
correct, we relaxed that requirement for the time being to see how this is actually used first. There still hasn't been any mobile app work lined up to use the values functionality.